builds/e2e/nested-e2e.yaml

```diff
@@ -110,13 +118,13 @@ stages:
           IotHubConnectionString: '$(IotHub-ConnStr)'
           test_type: nestededge_amqp
 
+- stage: Cleanup 
+  condition: always()
```

Comment:
I like that this cleanup stage is configured to always run. Would this be an appropriate place to also have a step that deletes iot hub device identities in case they get created and the test fails to complete? Or is this handled somewhere else such as within the identity service?  I've noticed a lot of devices in the Hub instances we use for testing and I'm wondering if there is a cleanup issue.

Reply:
There is a `Clean_up_identities` step that does exactly that a few lines below. It is also always configured to run.

You make a good point though, because the connectivity and longhaul pipelines don't clean up the identities. Ideally they should be imo. But there have never been any issues in longhaul. I know that for sure.

If we do want to make the change I can create a PBI for test improvement. Would rather have that be out of scope of this PR since the tests don't currently do that and it might not be causing any issues. If it turns up that connectivity module identity problems are related to too-many-deviceids we should definitely add this cleanup everywhere.

Reply:
Agree that it's out of scope for this PR but also agree that a PBI for the test improvement would be good to have.